components-api/src/main/scala/pl/touk/nussknacker/engine/api/typed/typing.scala

```diff
@@ -67,12 +69,29 @@ object typing {
 
   }
 
-  case class TypedTaggedValue(underlying: SingleTypingResult, tag: String) extends SingleTypingResult {
+  sealed trait TypedObjectWithData extends SingleTypingResult {
+    def underlying: SingleTypingResult
+    def data: Any
 
     override def objType: TypedClass = underlying.objType
+  }
 
+  case class TypedTaggedValue(underlying: SingleTypingResult, tag: String) extends TypedObjectWithData {
+    override def data: Any = tag
     override def display: String = s"${underlying.display} @ $tag"
+  }
 
+  case class TypedObjectWithValue(underlying: TypedClass, data: Any) extends TypedObjectWithData {
+    val maxDataDisplaySize = 15
+    val maxDataDisplaySizeWithDots = 12
```

Comment:
Can we have only one constant and compute e.g. this by `maxDataDisplaySize - "...".length`?